### PR TITLE
Fix the issue "Endless exception dialogues with illegal char in file path"

### LINKED ISSUE #7381: Endless exception dialogues with illegal char in file path
JabRef version 5.2 on Windows 10 (19042)

- [x] **Mandatory**: I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

1. Start with the file `test.bib`
    ```
    % Encoding: UTF-8
    
    @Article{test,
      file = {:C$\:User/someone/somefile.pdf:PDF},
    }
    ```
2. Open JabRef and Ctrl+O to open this file.
3. Java error is thrown. 

<details>
  <summary>Main problem here is, you are stuck in endless Java exception dialogs and has to terminate JabRef the hard way.</summary>
```
java.nio.file.InvalidPathException: Illegal char <:> at index 2: C$:User/someone/Baalbergen et al. - 2018 - Integrated collaboration capabilities for competitive aircraft design.pdf
	at java.base/sun.nio.fs.WindowsPathParser.normalize(Unknown Source)
	at java.base/sun.nio.fs.WindowsPathParser.parse(Unknown Source)
	at java.base/sun.nio.fs.WindowsPathParser.parse(Unknown Source)
	at java.base/sun.nio.fs.WindowsPath.parse(Unknown Source)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(Unknown Source)
	at java.base/java.nio.file.Path.of(Unknown Source)
	at org.jabref@5.2.60000/org.jabref.logic.importer.util.FileFieldParser.convert(Unknown Source)
	at org.jabref@5.2.60000/org.jabref.logic.importer.util.FileFieldParser.parse(Unknown Source)
	at java.base/java.util.Optional.map(Unknown Source)
	at com.tobiasdiez.easybind@2.1.0/com.tobiasdiez.easybind.optional.PreboundOptionalBinding$1.computeValue(Unknown Source)
	at com.tobiasdiez.easybind@2.1.0/com.tobiasdiez.easybind.optional.PreboundOptionalBinding$1.computeValue(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.beans.binding.ObjectBinding.get(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.beans.binding.ObjectExpression.getValue(Unknown Source)
	at com.tobiasdiez.easybind@2.1.0/com.tobiasdiez.easybind.optional.PreboundOptionalBinding$3.computeValue(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.beans.binding.ObjectBinding.get(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.beans.binding.ObjectExpression.getValue(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.TableCell.updateItem(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.TableCell.indexChanged(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.IndexedCell.updateIndex(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.skin.TableRowSkinBase.updateCells(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.skin.TableRowSkinBase.<init>(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.skin.TableRowSkin.<init>(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.TableRow.createDefaultSkin(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.Control.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.Control$1.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.control.ControlHelper.processCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.NodeHelper.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent$1.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.ParentHelper.processCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.NodeHelper.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent$1.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.ParentHelper.processCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.NodeHelper.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent$1.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.ParentHelper.processCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.control.ControlHelper.superProcessCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.control.ControlHelper.superProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.Control.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.Control$1.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.control.ControlHelper.processCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.NodeHelper.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent$1.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.ParentHelper.processCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.NodeHelper.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent$1.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.ParentHelper.processCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.control.ControlHelper.superProcessCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.control.ControlHelper.superProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.Control.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.Control$1.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.control.ControlHelper.processCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.NodeHelper.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent$1.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.ParentHelper.processCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.control.ControlHelper.superProcessCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.control.ControlHelper.superProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.Control.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.Control$1.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.control.ControlHelper.processCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.NodeHelper.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent$1.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.ParentHelper.processCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.NodeHelper.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Parent$1.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.ParentHelper.processCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.control.ControlHelper.superProcessCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.control.ControlHelper.superProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.Control.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.Control$1.doProcessCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.control.ControlHelper.processCSSImpl(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.scene.NodeHelper.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Node.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Node.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Node.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Node.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Node.processCSS(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Scene.doCSSPass(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.Scene$ScenePulseListener.pulse(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.tk.Toolkit.lambda$runPulse$2(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.tk.Toolkit.runPulse(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.tk.Toolkit.firePulse(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.tk.quantum.QuantumToolkit.pulse(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.tk.quantum.PaintCollector.liveRepaintRenderJob(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$ViewEventNotification.run(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$ViewEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleViewEvent$15(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleViewEvent(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.glass.ui.View.handleViewEvent(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.glass.ui.View.notifyResize(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.glass.ui.win.WinWindow._setBounds(Native Method)
	at org.jabref.merged.module@5.2.60000/com.sun.glass.ui.win.WinWindow.setBounds(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.tk.quantum.WindowStage.setBounds(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.stage.Window$TKBoundsConfigurator.apply(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.stage.Window.applyBounds(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.stage.Window.adjustSize(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.stage.Window.sizeToScene(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.stage.Window$12.invalidated(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.beans.property.BooleanPropertyBase.markInvalid(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.beans.property.BooleanPropertyBase.set(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.stage.Window.setShowing(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.stage.Window.show(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.stage.Stage.show(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.stage.Stage.showAndWait(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.HeavyweightDialog.showAndWait(Unknown Source)
	at org.jabref.merged.module@5.2.60000/javafx.scene.control.Dialog.showAndWait(Unknown Source)
	at org.jabref@5.2.60000/org.jabref.gui.JabRefDialogService.showErrorDialogAndWait(Unknown Source)
	at org.jabref@5.2.60000/org.jabref.gui.FallbackExceptionHandler.lambda$uncaughtException$0(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(Unknown Source)
	at org.jabref.merged.module@5.2.60000/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at org.jabref.merged.module@5.2.60000/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```
</details>

I tried it with the field `author` and it does not result in such error. Hence I reckon it should be related to the path parser.
This happens when I editted the entires manually.

### PR DESCRIPTION
**Pull request by Fu188**
If the user open a library including an illegal file path, the application will crash in the original version (endless exception dialogues). In this case, I add an exception handling to deal with this problem and prompt user "the file path is invalid". Fixes #7381 

- [x] Change in CHANGELOG.md described (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

**Testcase**

```bibtex
@InCollection{Favre2004,
Title = {{AVS and AVS/Express}},
Author = {Jean Favre and Mario Valle},
Booktitle = {{The Visualization Handbook}},
Publisher = {Academic Press},
Year = {2004},
Editor = {Chuck Hansen and Chris Johnson},
Month = dec,
Pages = {655--672},
File = {Paper:file:///D:\Books\Publications\ch33%20favre.pdf:PDF},
Keywords = {books},
Mvpubtype = {books},
Langid = {english},
Url = {http://www.elsevierdirect.com/product.jsp?isbn=9780123875822}
}
```

### CODE REVIEW

**Comment by calixtus** [comment]
Hi @Fu188, this seems to be the third PR with the same commit stream. Please do not close and open multiple PRs for the same commit stream, since the new PR page does not show the old comments and remarks we made in the commit history and it pollutes the issue tracker.
Please also stay patient, our policy is to always review a PR by at least two core developers before merging it into the master branch. Since we all work on JabRef in our free time, sometimes this can take a little time.

Refs #7388 

Right now, the tests are failing:
```
 /home/runner/work/jabref/jabref/src/main/java/org/jabref/logic/importer/util/FileFieldParser.java:60: error: cannot find symbol
        if (!entry.isEmpty()) {
             ^
  symbol:   variable entry
  location: class FileFieldParser
/home/runner/work/jabref/jabref/src/main/java/org/jabref/logic/importer/util/FileFieldParser.java:61: error: cannot find symbol
            LinkedFile linkedFile = convert(entry);
                                            ^
  symbol:   variable entry
  location: class FileFieldParser
```

File: CHANGELOG.md
Comment (koppor) [inline CHANGELOG.md:43]: Please do not delete existing changelog entries. The "trick" is to add the change in a random position in the list (and not to the end)

I readded it in this suggestion:

```suggestion
- We fixed an issue where the file path is invisible in dark theme. [#7382](https://github.com/JabRef/jabref/issues/7382)
- We fixed an issue with endless exception dialogues appearing if the file path includes an illegal char. [#7381](https://github.com/JabRef/jabref/issues/7381)
```

File: src/main/java/org/jabref/logic/importer/util/FileFieldParser.java
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/util/FileFieldParser.java:64]: We ❤️ Java 8, thus following suggestion

```suggestion
            Optional.ofNullable(convert(linkedFileData)).ifPresent(files::add);
```

File: src/main/java/org/jabref/logic/importer/util/FileFieldParser.java
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/util/FileFieldParser.java:105]: This is sort of a quick hack. There is no loging happening, no output to the user, nothing.

I think, the invalid path is just thrown away. This should not happen. The invalid path should still be displayed in the UI. Possible, the path should be fixed. This should be easy in your case, because, the `file:///` can just be removed, can't it? There will be special handling required to handle the case `:/` when splitting the list for Linkedfile

(The issue is that x:file:///thing:pdf will be x, file, ///thing:pdf as list and not x, file:///thing, pdf. Please think of the parsing more instead of adding a quick fix. This will help many users manually editing the bibtex and hoping that JabRef does not kill data.

**Review by koppor — CHANGES_REQUESTED** [review]
Please add test cases for the changes. Otherwise, I cannot really get what happens. Detailed comments inside.

**Comment by calixtus** [comment]
Devcall: What's the status here?

**Comment by calixtus** [comment]
Devcall: Since there was no reaction by the author and this PR seems to need more refactoring, we decided to freeze and close this PR.

**Pull request closed without merging**
